### PR TITLE
new commitQueue

### PR DESCRIPTION
**Pull request by 132yse**
1. Remove the while loop of the linked list instead of forEach of the array.
2. Use global commitQueue

This PR improves performance, it also simplifies the logic of the commit phase. The commitWork is synchronous, which is not our focus. It should not be too complex.

### CODE REVIEW

File: test/render.test.jsx [hunk 1]
```diff
@@ -116,12 +116,6 @@ test('render/update object properties and DOM attributes', async () => {
 
         lastChildren.forEach((lastChild, index) => expect(elements[0].children[index]).toBe(lastChild))
       }
-    },
-    {
-      content: "removed",
-      test: ([text]) => {
-        expect(text.textContent).toBe("removed")
-      }
```
Comment (mindplay-dk) [inline test/render.test.jsx]: Why did you remove this step of the test?

Comment (132yse) [reply test/render.test.jsx]: Sorry, because I rolled back the code, I'll add it again later.
